addons/website_cf_turnstile/static/src/js/turnstile.js

```diff
@@ -4,52 +4,81 @@ import "@website/snippets/s_website_form/000";  // force deps
 import publicWidget from '@web/legacy/js/public/public_widget';
 import { session } from "@web/session";
 
+export const turnStile = {
+    addTurnstile: function (action) {
+        if (!this.isEditable && !this.$('.s_turnstile').length && session.turnstile_site_key) {
+            const mode = new URLSearchParams(window.location.search).get('cf') == 'show' ? 'always' : 'interaction-only';
+            return $(`<div class="s_turnstile cf-turnstile float-end"
+                        data-action="${action}"
+                        data-appearance="${mode}"
+                        data-response-field-name="turnstile_captcha"
+                        data-sitekey="${session.turnstile_site_key}"
+                        data-error-callback="throwTurnstileError"
+                ></div>
+                <script class="s_turnstile">
+                    // Rethrow the error, or we only will catch a "Script error" without any info
+                    // because of the script api.js originating from a different domain.
+                    function throwTurnstileError(code) {
+                        const error = new Error("Turnstile Error");
+                        error.code = code;
+                        throw error;
+                    }
+                </script>
+                <script class="s_turnstile" src="https://challenges.cloudflare.com/turnstile/v0/api.js"></script>
+            `);
+        }
+    },
+
+    /**
+     * Remove potential existing loaded script/token
+     */
+    cleanTurnstile: function () {
+        if (this.$('.s_turnstile').length) {
+            this.$('.s_turnstile').remove();
+        }
+    },
+
+    /**
+     * @override
+     * Discard all library changes to reset the state of the Html.
+     */
+    destroy: function () {
+        this.cleanTurnstile();
+        this._super(...arguments);
+    },
+};
+
+const signupTurnStile = {
+    ...turnStile,
+    async willStart() {
+        this._super(...arguments);
+        this.cleanTurnstile();
+        this.addTurnstile(this.action)?.insertBefore(`${this.selector} button[type='submit']`);
```

Comment:
So, I would change that
```javascript
`${this.selector} button[type='submit']`
```
to
```javascript
this.el.querySelector("button[type='submit']")
```
as it would be more correct (ensuring you only consider the *unique* button inside the *current* element matched by this.selector, even though in this case, it does not matter) and more efficient (looking for the inner element instead of the whole DOM, although negligeable).

Reply:
I agree with Quentin to put `this.el.querySelector("button[type='submit']")`

Reply:
If there are multiple forms, your public widget using your mixin will be instantiated multiple times. So if there are multiple forms matching your selector, it would be definitely wrong not to apply my suggestion as you would "addTurnstile" a squared number of times. If there is only one form, this is still better, as mentioned before.